components/esp32/panic.c

```diff
@@ -228,6 +229,32 @@ static void disableAllWdts() {
 
 #endif
 
+static inline bool stackPointerIsSane(uint32_t sp) {
+	return !(sp < 0x3ffae010 || sp > 0x3ffffff0 || ((sp & 0xf) != 0));
+}
+static void putEntry(uint32_t pc, uint32_t sp) {
+	char buf[24];
+	if (pc & 0x80000000) pc = (pc & 0x3fffffff) | 0x40000000;
+	sprintf(buf, " %p:%p", (void *) pc, (void *) sp);
```

Comment:
done. i made a couple minor changes: panicPutchar -> panicPutChar (sorry, OCD :)) and also added `0x` to hex numbers - both addr2line and gdb take hex numbers with 0x, so this is more copy-paste friendly.